source/AI.cpp

```diff
@@ -3142,6 +3220,28 @@ double AI::RendezvousTime(const Point &p, const Point &v, double vp)
 }
 
 
+// Searches every asteroid within the ship scan limit and returns the asteroid
+// closest to the ship.
+bool AI::TargetAsteroid(Ship &ship) const
+{
+	double scanLimit = 100. * sqrt(ship.Attributes().Get("asteroid scan power"));
+	if(!scanLimit)
+	{
+		return false;
+	}
+	for(const shared_ptr<Minable> &asteroid : minables)
+	{
+		double range = ship.Position().Distance(asteroid->Position());
+		if(range < scanLimit)
+		{
+			ship.SetTargetAsteroid(asteroid);
+			scanLimit = range;
```

Comment:
Worth noting I did not create this logic.  I moved it from another part of the game code.  Here it is in vanilla

https://github.com/endless-sky/endless-sky/blob/d7b158c96fa65d3bd933cd5c694055737cc3a043/source/AI.cpp#L3271-L3283

This would change the asteroid scanning behavior (I do not know how it would affect gameplay).  Are you sure you want to remove sqrt?  We might consider doing this change outside of my PR because it could potentially break asteroid scanning in general.

Reply:
Why would it change the behavior if we compare the squared metric with the squared distance, vs the square root of the metric vs the not-squared distance? Distance is not signed.

Reply:
You can install 10x asteroid scanners.  Right now that would be `sqrt(10x)` range.  Your change makes it a full 10x for range increase.  I'm not against better range just that I haven't play tested with this yet.

Reply:
```
X = 10
Y = 16
X^2 * Y == 1600
X * Y^(1/2) == 40
```

This is a pretty drastic difference.  Again, not that I _really care_ since it makes asteroid scanning a lot better.

Reply:
> ```
> X = 10
> Y = 16
> X^2 * Y == 1600
> X * Y^(1/2) == 40
> ```
> 
> This is a pretty drastic difference.  Again, not that I _really care_ since it makes asteroid scanning a lot better.  I just don't know if it's good for it to be mixed with my PR because it is a pretty core logic change to how both player ships and AI behave.


Original code compares A with B. My suggested code compares A² with B². Getting A and B, vs A² and B², requires two calls to sqrt.

Reply:
This change appears to have broken asteroid scanning by making it essentially 400k.

Reply:
After discussion in discord I didn't notice `DistanceSquared` vs `Distance`.  The math checks out I just didn't see that second part.